### ISSUE TITLE
Page references in directives not updated on rename

### ISSUE THREAD

**Issue by ishehadeh** [silverbulletmd/silverbullet#462]
When a page name changes silverbullet won't update references inside directives. This is mostly (entirely?) an issue for `render` statements in query directives.

Steps to reproduce:

- Add a page `template/rename-test`, with text "- {{name}}"
- Add a page with a query `<!-- #query page render [[template/rename-test]] --><!-- /query -->`
- rename `template/rename-test` to `template/rename-test2`.
- query won't change

**Comment by zefhemel** [comment]
Yeah this is somewhat on purpose. The rename functionality uses the wiki page backlink index, which doesn't index links inside directives: https://github.com/silverbulletmd/silverbullet/blob/main/plugs/core/page.ts#L36

As a result, these links won't be updated. So far I've been ok with it, because this fixes itself the moment a directive re-renders itself, but it's not super great. Perhaps I'll change how backlink indexing works and add links inside directives too. I'm sure I had a good reason for it at the time, but honestly I don't remember, so perhaps simply removing this line will be fine :D
